functional_test.go

```diff
@@ -228,7 +228,7 @@ mainLoop:
 			}
 			for _, broker := range brokers {
 				err := broker.Open(client.Config())
-				if err != nil {
+				if err != nil && err != ErrAlreadyConnected {
```

Comment:
Updated the functional tests to skip the ErrAlreadyConnected error because this change will use one broker from the broker list to refresh metadata, which is already connected when reaching the following validation.